### PR TITLE
Implement liveTx and min/max height search filters

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: packages/bcp/src/cosmwasmconnection.ts [hunk 1]
```diff
@@ -314,30 +314,38 @@ export class CosmWasmConnection implements BlockchainConnection {
...
     } else if (sentFromOrTo) {
+      // TODO: pass minHeight/maxHeight to server once we have
+      // https://github.com/cosmwasm/wasmd/issues/73
       txs = await this.cosmWasmClient.searchTx({ sentFromOrTo: sentFromOrTo });
     } else {
       throw new Error("Unsupported query");
     }
 
-    return txs.map(tx => this.parseAndPopulateTxResponseUnsigned(tx));
+    const filtered = txs.filter(tx => {
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmconnection.ts]: Oof, filtering client side. "Fake it til you make it".

But yeah, I agree you can fulfill the API properly now, while not being blocked on the "proper" backend changes.

File: packages/bcp/src/cosmwasmconnection.ts [hunk 2]
```diff
@@ -347,9 +355,55 @@ export class CosmWasmConnection implements BlockchainConnection {
...
+    if ([query.height, query.signedBy, query.tags].some(isDefined)) {
+      throw new Error("Transaction query by height, signedBy or tags not yet supported");
+    }
+
+    if (query.id) {
+      if (query.minHeight || query.maxHeight) {
+        throw new Error("Query by minHeight/maxHeight not supported together with ID");
+      }
+
+      // concat never() because we want non-completing streams consistently
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmconnection.ts]: interesting

File: packages/sdk/src/restclient.ts [hunk 3]
```diff
@@ -152,7 +152,19 @@ type RestClientResponse =
   | WasmResponse<string>
   | WasmResponse<GetCodeResult>;
 
-type BroadcastMode = "block" | "sync" | "async";
+/**
+ * The mode used to send transaction
+ *
+ * @see https://cosmos.network/rpc/#/Transactions/post_txs
+ */
+export enum BroadcastMode {
```
Comment (ethanfrey) [inline packages/sdk/src/restclient.ts]: Good docs

File: packages/bcp/src/cosmwasmconnection.ts (hunk 1 above)
Comment (webmaster128) [reply packages/bcp/src/cosmwasmconnection.ts]: Exactly. But feel free to bring in https://github.com/cosmwasm/wasmd/issues/73

**Pull request merged**